### PR TITLE
Add sidebar activity errors and empty state

### PR DESCRIPTION
**Pull request by fonurr**

### CODE REVIEW

**Review by konukselim — APPROVED** [review]

**Comment by fonurr** [comment]
Rebased

File: src/sidebar/activity/tree/ActivityTreeDataProvider.ts [hunk 1]
```diff
@@ -20,8 +23,9 @@ class ActivityTreeDataProvider extends TreeDataProvider {
     }
 
     public async getRoots(): Promise<TreeItem[]> {
-        // TODO: Handle errors
-        const { data } = await ActivityStore.get();
+        const { data, errors } = await ActivityStore.get();
+
+        const roots: TreeItem[] = errors?.length ? [new ActivityErrorsTreeItem(errors)] : [];
```
Comment (mertkahyaoglu) [inline src/sidebar/activity/tree/ActivityTreeDataProvider.ts:28]: I think we should move this below activities calculation where we push into it.

File: src/sidebar/activity/tree/ActivityTreeDataProvider.ts [hunk 2]
```diff
@@ -32,8 +36,12 @@ class ActivityTreeDataProvider extends TreeDataProvider {
             }
             slots[slot].push(activity);
         });
-        const slotTreeItems = Object.keys(slots).map(key => new ActivitySlotTreeItem(key as DateSlot, slots[key]));
-        return slotTreeItems;
+
+        roots.push(...Object.keys(slots).map(key => new ActivitySlotTreeItem(key as DateSlot, slots[key])));
```
Comment (mertkahyaoglu) [inline src/sidebar/activity/tree/ActivityTreeDataProvider.ts:40]:
```suggestion
        roots.concat(Object.keys(slots).map(key => new ActivitySlotTreeItem(key as DateSlot, slots[key])));
```

**Review by mertkahyaoglu — APPROVED** [review]

File: src/sidebar/activity/tree/ActivityTreeDataProvider.ts (hunk 1 above)
Comment (fonurr) [reply src/sidebar/activity/tree/ActivityTreeDataProvider.ts:28]: Error is the first item we show, so I calculated it first

File: src/sidebar/activity/tree/ActivityTreeDataProvider.ts (hunk 2 above)
Comment (fonurr) [reply src/sidebar/activity/tree/ActivityTreeDataProvider.ts:40]: I would have to reassign your suggestion to roots. So I would like it to remain this way.

`roots = roots.concat(Object.keys(slots).map(key => new ActivitySlotTreeItem(key as DateSlot, slots[key])));`

 Am I wrong?

**Comment by fonurr** [comment]
I am merging thi. I will open a new PR if there is anything that needs to be changed comes up on comments.

**Pull request merged**